pkg/cmd/server/origin/controller/autoscaling.go

```diff
@@ -44,14 +38,30 @@ func (c *HorizontalPodAutoscalerControllerConfig) RunController(originCtx Contro
 		"heapster",
 		"",
 	)
-	replicaCalc := hpacontroller.NewReplicaCalculator(metricsClient, hpaClient.Core())
+	replicaCalc := hpacontroller.NewReplicaCalculator(
+		metricsClient,
+		hpaClient.CoreV1(),
+		0.1, // this is the default
+	)
 
-	delegatingScalesGetter := appsv1client.NewDelegatingScaleNamespacer(appsClient, hpaClient.ExtensionsV1beta1())
+	// TODO: we need something like deferred discovery REST mapper that calls invalidate
+	// on cache misses.
+	cachedDiscovery := discocache.NewMemCacheClient(hpaClient.Discovery())
+	restMapper := discovery.NewDeferredDiscoveryRESTMapper(cachedDiscovery, apimeta.InterfacesForUnstructured)
+	restMapper.Reset()
+	// we don't use cached discovery because DiscoveryScaleKindResolver does its own caching,
+	// so we want to re-fetch every time when we actually ask for it
+	scaleKindResolver := scale.NewDiscoveryScaleKindResolver(hpaClient.Discovery())
+	scaleClient, err := scale.NewForConfig(hpaClientConfig, restMapper, dynamic.LegacyAPIPathResolverFunc, scaleKindResolver)
```

Comment:
🎉

Reply:
actually, do we even need any of this code at all?  If we pull the trigger on metrics-server instead of Heapster for the HPA, we can drop all of our custom setup.  I'm not sure how we want to handle that, though...